packages/arb-token-bridge-ui/src/state/app/actions.ts

```diff
@@ -30,19 +27,17 @@ export const setSelectedToken = (
 
 export const reset = ({ state }: Context, newChainId: number) => {
   if (
-    state.app.l1NetworkChainId !== newChainId &&
-    state.app.l2NetworkChainId !== newChainId
+    state.app.sourceChainId !== newChainId &&
+    state.app.destinationChainId !== newChainId
   ) {
     // only reset the selected token if we are not switching between the pair of l1-l2 networks.
     // we dont want to reset the token if we are switching from Goerli to Arbitrum Goerli for example
     // because we are maybe in the process of auto switching the network and triggering deposit or withdraw
     state.app.selectedToken = null
   }
```

Comment:
i think this condition causes the bug we had with selected token (which is hackily fixed now by resetting the selected token to `null` whenever we change source/destination chain id in `TransferPanelMain`)

here with this code snippet, when user's source is Ethereum, and destination is Arb One, and then user changes destination chain to Arb Nova, the source chain id === new chain id, so the selected token remains the same but the token data should have been different